dcpy/models/product/dataset/metadata_v2.py

```diff
@@ -0,0 +1,424 @@
+from __future__ import annotations
+
+import jinja2
+
+from pathlib import Path
+from pydantic import BaseModel, field_validator, model_serializer
+from pydantic.fields import PrivateAttr
+from typing import Any, List, Literal, get_args, Self
+import typing
+import unicodedata
+import yaml
+
+from dcpy.utils.logging import logger
+from dcpy.utils.collections import deep_merge_dict as merge
+
+
+# MISC UTILS
+def normalize_text(s):
+    """
+    Normalize the text we may receive from the various metadata sources.
+    Primarily useful for cleaning long-text like descriptions.
+    """
+    char_map = {
+        "–": "-",  # en dash to hyphen
+        "—": "-",  # em dash to hyphen
```

Comment:
duplicated?

Reply:
I think the font isn't doing us any favors here, but one is U+2013 and the other is U+2014. Lots of this stuff came from copying in from pdfs.